### PR TITLE
Fix font axes/features not working on DPI change

### LINKED ISSUE #11287: Font Features is inconsistent
### Windows Terminal version (or Windows build number)

10.0.19043.1237, 1.11.2421.0

### Other Software

_No response_

### Steps to reproduce

I'm using the Nerd-Font port of FiraCode as my font for Windows Terminal. Using the new Font Features options, I've configured WT to use the alternate zero glyph which replaces the slash in the middle of the character with a dot. There have been a couple instances in which I've launched a terminal only for the zero glyph to have reverted back to the old style, though launching a new terminal does fix the issue.

Here is my settings config:
```
"font": 
{
  "face": "FiraCode NF",
  "features": 
  {
    "calt": 0,
    "ss04": 1,
    "zero": 1
  },
  "size": 12
},
```

~~I think I've found a way to reproduce this. It seems like it happens when you first launch windows terminal from a full restart.~~
Ignore that last one, it doesn't happen all the time. I can trigger the effect by dragging my window from a display at 100% scaling to one with 125% scaling.

### Expected Behavior

The font should display with the correct `0` symbol

### Actual Behavior

The zero symbol is incorrect.

### PR DESCRIPTION
**Pull request by PankajBhojwani**
When the dpi is changed, call `updateFont()` instead of `TriggerFontChange`, this
means that we continue to use the existing font features/axes

Closes #11287

### CODE REVIEW

File: src/renderer/dx/DxFontRenderData.cpp [hunk 1]
```diff
@@ -178,12 +178,46 @@ DxFontRenderData::DxFontRenderData(::Microsoft::WRL::ComPtr<IDWriteFactory1> dwr
...
+        _defaultFontInfo = DxFontInfo(desired.GetFaceName(),
+                                      desired.GetWeight(),
+                                      DWRITE_FONT_STYLE_NORMAL,
+                                      DWRITE_FONT_STRETCH_NORMAL);
+
+        _BuildFontRenderData(desired, actual, dpi);
+    }
+    CATCH_RETURN();
+
+    return S_OK;
```
Comment (PankajBhojwani) [inline src/renderer/dx/DxFontRenderData.cpp:208]: I don't like that this is exactly the same as the other `UpdateFont` only without the `_SetFeatures` and `_SetAxes` calls

File: src/renderer/dx/DxRenderer.cpp [hunk 2]
```diff
@@ -1987,7 +1987,18 @@ CATCH_RETURN()
...
+        // If no font features or axes were supplied, we call the version of UpdateFont that
+        // tells the DxFontRenderData to continue using the features/axes it already has (if any)
+        RETURN_IF_FAILED(_fontRenderData->UpdateFont(pfiFontInfoDesired, fiFontInfo, _dpi));
+
+        // Prepare the text layout.
+        _customLayout = WRL::Make<CustomTextLayout>(_fontRenderData.get());
+
+        return S_OK;
+    }
+    CATCH_RETURN();
```
Comment (PankajBhojwani) [inline src/renderer/dx/DxRenderer.cpp:2001]: I also don't like that this is almost exactly the same as the other `UpdateFont` in this file either

**Review by zadjii-msft — APPROVED** [review]

File: src/renderer/dx/DxFontRenderData.cpp (hunk 1 above)
Comment (zadjii-msft) [reply src/renderer/dx/DxFontRenderData.cpp:208]: meh, I'm not sure how to make it better. A `const std::optional<std::unordered_map<std::wstring_view, uint32_t>>& features` is disgusting

**Comment by PankajBhojwani** [comment]
@lhecker 's suggestion makes a lot more sense, PR has been updated to use `updateFont()` instead (and confirmed that the fix works)

File: src/cascadia/TerminalControl/ControlCore.cpp [hunk 3]
```diff
@@ -924,9 +924,7 @@ namespace winrt::Microsoft::Terminal::Control::implementation
         auto lock = _terminal->LockForWriting();
         _compositionScale = scale;
 
-        _renderer->TriggerFontChange(::base::saturated_cast<int>(dpi),
-                                     _desiredFont,
-                                     _actualFont);
+        _updateFont();
```
Comment (DHowett) [inline src/cascadia/TerminalControl/ControlCore.cpp:927]:
```suggestion
        // _updateFont relies on the new _compositionScale set above
        _updateFont();
```

I had to look into the code to see this, but this is only a _nit_ :)

**Review by DHowett — APPROVED** [review]

File: src/cascadia/TerminalControl/ControlCore.cpp (hunk 3 above)
Comment (PankajBhojwani) [reply src/cascadia/TerminalControl/ControlCore.cpp:928]: Makes sense! Added

**Comment by lhecker** [comment]
@PankajBhojwani This fixes #11317 now, which is why I've modified your commit message.

**Comment by DHowett** [comment]
From chat:

nevermind, it doesn't fix it ;P

**Pull request merged**